### PR TITLE
Merge master into pt

### PR DESCRIPTION
**Pull request by karussell**
This brings the PT branch to master. All tests pass but please review before, especially the clashing classes like Dijkstra and Path

> Aaah - this is VERY confusing that all the **old** comments regarding some containing commits now show up in this pull request!!

Okay, squashed all changes into a single commit for now

### CODE REVIEW

File: core/src/main/java/com/graphhopper/routing/Path.java [hunk 1]
```diff
@@ -214,7 +217,7 @@ public void next(EdgeIteratorState edgeBase, int index) {
                         // TODO This should already be in the SPT, we shouldn't need to calculate it again here.
                         time += ((TimeDependentWeighting) weighting).calcTravelTimeSeconds(edgeBase, time / 1000.0) * 1000.0;
                     } else {
-                        time += calcMillis(dist, edgeBase.getFlags(), false);
+                        time += weighting.calcMillis(edgeBase, false, EdgeIterator.NO_EDGE);
```
Comment (karussell) [inline core/src/main/java/com/graphhopper/routing/Path.java]: Here we need previousEdge too

**Comment by karussell** [comment]
(GitHub going nuts now when I edit a comment it shows up a previous version ... anyway)

We need to work out a solution for #831 with public transport in mind.

**Comment by karussell** [comment]
I've fixed the issues and also merged latest changes from pt branch back to here so it should be mergable without conflicts again. Would you mind to review @michaz ?

**Comment by michaz** [comment]
Thanks! Quite possible that some of the changes in the core on the pt branch can be reverted and left in a new branch (time-dependent) soon.

**Comment by karussell** [comment]
Probably we can also consider to merge pt into master as soon as it is at least 'alpha' status (e.g. slightly tuned web API&UI and a bit more unit test coverage) because 0.9 has still several months to stabilize & evolve. And maintaining two not so similar branches requires a bit work.

**Pull request closed without merging**